File: Calculator.Runtime.Java/src/main/java/calculator/runtime/java/actors/CalculatorActor.java
package calculator.runtime.java.actors;

import calculator.runtime.java.models.Calculation;
import io.dapr.actors.ActorMethod;
import io.dapr.actors.ActorType;
import reactor.core.publisher.Mono;

@ActorType(name = "CalculatorActor")
public interface CalculatorActor {
    @ActorMethod(name = "Execute", returns = Object.class)
    Mono<Object> execute(Calculation calculation);
}

File: MyRedisPusSub/Calculator.Runtime.Dotnet/Actors/CalculatorActor.cs
using Calculator.Runtime.Dotnet.Models;
using Calculator.Runtime.Dotnet.Services;
using Dapr.Actors.Runtime;

namespace Calculator.Runtime.Dotnet.Actors;

public class CalculatorActor : Actor, ICalculatorActor
{
    private readonly ICalculationService _calculationService;
    private readonly IParametersTransformationService _transformationService;
    private readonly ILogger<CalculatorActor> _logger;

    public CalculatorActor(ActorHost host,
        ICalculationService calculationService,
        IParametersTransformationService transformationService,
        ILogger<CalculatorActor> logger) : base(host)
    {
        _calculationService = calculationService;
        _transformationService = transformationService;
        _logger = logger;
    }

    public async Task<object> Execute(Calculation calculation)
    {
        _logger.LogInformation("Received: {calculation}", calculation);
        _logger.LogInformation("Received: {parameters}", calculation.Parameters);

        var parameters = _transformationService.Transform(calculation.Parameters);
        var command = new CalculationCommand(calculation.Expression, parameters);

        var result = await _calculationService.ExecuteAsync(command);

        return result;
    }
}